tools/check/check.go

```diff
@@ -129,33 +200,211 @@ func die(err error) {
 	dieWith(err, exitTool)
 }
 
+func parseConfig(path string) error {
+	if path == "" {
+		return nil
+	}
+	f, err := os.Open(path)
+	if err != nil {
+		return fmt.Errorf("could not open %q: %v", path, err)
+	}
+	defer f.Close()
+
+	contents, err := io.ReadAll(f)
+	if err != nil {
+		return fmt.Errorf("could not read %q: %v", path, err)
+	}
+	if strings.HasSuffix(path, ".textproto") {
+		err = prototext.Unmarshal(contents, config)
+	} else {
+		err = proto.Unmarshal(contents, config)
+	}
+	if err != nil {
+		return fmt.Errorf("could not deserialize %q: %v", path, err)
+	}
+	// Populate fields that should not be nil
+	if config.RootOfTrust == nil {
+		config.RootOfTrust = &checkpb.RootOfTrust{}
+	}
+	if config.Policy == nil {
+		config.Policy = &checkpb.Policy{HeaderPolicy: &checkpb.HeaderPolicy{}, TdQuoteBodyPolicy: &checkpb.TDQuoteBodyPolicy{}}
+	}
+	return nil
+}
+
+func parsePaths(s string) ([]string, error) {
+	paths := strings.Split(s, ":")
+	if len(paths) == 1 && paths[0] == "" {
+		return nil, nil
+	}
+	var result []string
+	for _, path := range paths {
+		p := strings.TrimSpace(path)
+		stat, err := os.Stat(p)
+		if err != nil {
+			return nil, fmt.Errorf("path error for %q: %v", p, err)
+		}
+		if stat.IsDir() {
+			return nil, fmt.Errorf("path is not a file: %q", p)
+		}
+		result = append(result, p)
+	}
+	return result, nil
+}
+
+func parseRtmrs(s string) ([][]byte, error) {
+	hexstrings := strings.Split(s, ",")
+	if len(hexstrings) == 1 && hexstrings[0] == "" {
+		return nil, nil
+	}
+	var result [][]byte
+	for _, hexstring := range hexstrings {
+		h, err := hex.DecodeString(strings.TrimSpace(hexstring))
+		if err != nil {
+			return nil, fmt.Errorf("could not parse RTMRS value as hex-encoded string: %q", hexstring)
+		}
+		result = append(result, h)
+	}
+	return result, nil
+}
+
+func setBool(value *bool, name, flag string, defaultValue bool) error {
+	if flag == "" {
+		if !override() {
+			*value = defaultValue
+		}
+	} else if flag == "true" {
+		*value = true
+	} else if flag == "false" {
+		*value = false
+	} else {
+		return fmt.Errorf("flag -%s=%s invalid. Must be one of unset, \"true\", or \"false\"",
+			name, flag)
+	}
+	return nil
+}
+
+func setUint(value *uint64, bits int, name, flag string, defaultValue uint64) error {
+	if flag == "" {
+		if !override() {
+			*value = defaultValue
+		}
+	} else {
+		u, err := parseUint(flag, bits)
+		if err != nil {
+			return fmt.Errorf("invalid -%s=%s: %v", name, flag, err)
+		}
+		*value = u
+	}
+	return nil
+}
+
+func setUint32(value *uint32, name, flag string, defaultValue uint64) error {
+	v := uint64(*value)
+	if err := setUint(&v, 32, name, flag, defaultValue); err != nil {
+		return err
+	}
+	*value = uint32(v)
+	return nil
+}
+
+func override() bool {
```

Comment:
It's hard to tell what this function does from the name. Can you use a more specific function name?

Reply:
Renamed to configProtoPresent()